frontend/src/components/PayLightningInvoice.tsx

```diff
@@ -1,18 +1,28 @@
 import { Invoice, fiat } from "@getalby/lightning-tools";
 import { CopyIcon, LightbulbIcon } from "lucide-react";
-import React from "react";
+import React, { useState } from "react";
 import { LightningIcon } from "src/components/icons/Lightning";
 import Loading from "src/components/Loading";
 import QRCode from "src/components/QRCode";
 import { Button, ExternalLinkButton } from "src/components/ui/button";
+import { LoadingButton } from "src/components/ui/loading-button";
 import { useToast } from "src/components/ui/use-toast";
 import { copyToClipboard } from "src/lib/clipboard";
+import useChannelOrderStore from "src/state/ChannelOrderStore";
+import { LSPOrderResponse } from "src/types";
+import { request } from "src/utils/request";
 
 type PayLightningInvoiceProps = {
   invoice: string;
+  lspOrderResponse?: LSPOrderResponse;
+  canPayInternally?: boolean | undefined;
 };
 
-export function PayLightningInvoice({ invoice }: PayLightningInvoiceProps) {
+export function PayLightningInvoice({
```

Comment:
This was a general "Pay Lightning Invoice" component and now it is coupled with the LSP order. This doesn't seem to make sense to me, can you double-check your changes?

Reply:
remove canPayInternally check. can be calculated from invoice amount and spendible balance